src/task_manager.cc

```diff
@@ -43,6 +44,16 @@ namespace detail {
 		}
 	}
 
+	void task_manager::notify_horizon_executed(task_id tid) {
```

Comment:
I just noticed that this assumes it is never called from 2 threads in parallel. If that can happen, we'd need a lock for the `executed_horizons` data structure.

Reply:
That's fine, worker jobs all run in the same thread!